### DISCUSSION TITLE
NO helpful error when generating sign url

### DISCUSSION THREAD

**Discussion by airyland** [phantom/docs#285] [Help & Support]
### Summary

I followed the doc(https://docs.phantom.app/phantom-deeplinks/provider-methods/signmessage) and did many tests to generate sign url, but all failed: `unexpected error`

``` mjs
import { Buffer } from 'buffer'
import { URLSearchParams } from 'url'
import bs58 from 'bs58'
import nacl from 'tweetnacl'

// for test
const config = {
  session: 'KDCgMju6D1HV8fWe298okbih1JEwaPrhxQTfQtSMAYJDD6wsTVhgvocPenUWUqU3grSGSs9hs9s19Y4gbkitj15cbu8BxmmMvTkW8sxAosLVvFHeokFxALtNiw4y4dA3i8XQtLFCi6KDj51eHYR7S5zFBXreFMCDwPU99ZjxDq1G9dFK8oN74QLPM7qHjNYh7SCtmzWDsZhyTU6t31BSCHPLa3L8pAcpqSK29DBvm6bc1VCU2LJTdVbPbNjn2vG',
  sec: '5SLoWKopvJgGMDA6Xx46fU1YpErBFZytxuAJUphLrYKF',
  dappEncryptionPublicKey: 'Cfo9u5Fzs6hBCEWgtU99L5HdjTzSJAb5gJRuaDJkfL1G',
  message: 'address: 65HZW83WNWjURRbxPQ1yVRXQhyquJv831Lui4XYj55Gi',
  timestamp: 1725413435251
}

const signMessage = async () => {
  const message = 'test message'

  const payload = {
    message: bs58.encode(Buffer.from(message)),
    session: config.session,
    display: 'utf8'
  }

  const publicKey = bs58.decode(config.dappEncryptionPublicKey)
  const secretKey = bs58.decode(config.sec)

  // Create the key pair object
  const dappKeyPair = {
    publicKey,
    secretKey
  }

  console.log(dappKeyPair)

  const sharedSecret = nacl.box.before(
    publicKey,
    secretKey
  )

  const [
    nonce,
    encryptedPayload
  ] = encryptPayload(payload, sharedSecret)

  const params = new URLSearchParams({
    dapp_encryption_public_key: bs58.encode(dappKeyPair.publicKey),
    nonce: bs58.encode(nonce),
    redirect_link: 'https://www.google.com',
    payload: bs58.encode(encryptedPayload),
    session: config.session,
    message: bs58.encode(Buffer.from(message)),
    display: 'utf8'
  })

  const url = buildUrl('signMessage', params)
  console.log('url', url)
}

const useUniversalLinks = false

const buildUrl = (path, params) =>
  `${useUniversalLinks ? 'https://phantom.app/ul/' : 'phantom://'}v1/${path}?${params.toString()}`

const encryptPayload = (payload, sharedSecret) => {
  if (!sharedSecret) throw new Error('missing shared secret')

  const nonce = nacl.randomBytes(24)

  const encryptedPayload = nacl.box.after(
    Buffer.from(JSON.stringify(payload)),
    nonce,
    sharedSecret
  )

  return [
    nonce,
    encryptedPayload
  ]
}

console.log(signMessage())


```

Could anyone help?

### Steps to Reproduce

Just exec the code, got the url and visit it in safari, then got the error:

`?errorCode=-32603&errorMessage=Unexpected+error`

url example: `phantom://v1/signMessage?dapp_encryption_public_key=Cfo9u5Fzs6hBCEWgtU99L5HdjTzSJAb5gJRuaDJkfL1G&nonce=2hzGV3k6P4VQHvZVKfjaojbYmsmcSMBCq&redirect_link=https%3A%2F%2Fwww.google.com&payload=3ff1VykJFtFcBoMzxRKUhBa4LMs3fhsasLxmnfr1SavRXuVoA35UTPUoGmfmfNMdYNctDkJxyw54zHM8xH1VYhMKBRzAFrWuu2mGGBG7Ro7UeGc5XGeqVGqvnfg3pqciiFt747zz5Fe8udYuCxPDnznL5e57KyTEwJZpTqpgS4iAdte6WvNdZWRueRbg293U87mP5vmUoy3HPc9zNyERjCfU69rbvqPYz7e5KQLKGmo5BhBzE8bdBMnXYyfUEjFhsDGGxY5XqYbiyHfe27gZMTbDkZbJxTamDufJPJ7n1LhqZuDze5TcMoUcgv6Ceqc3J3sZtEUjMqVGVLftNXCS7j4ZrUhVFnLhyQX6n32Lasy6ovwkQWjjnGAYWLbNUbb6P7q9ZuBbgniKP3gEmCpVSd6eAwQUHP5g42B98v7VATk2vFxTEwmpn5PX3MGfX6Vb8dTRMM&session=KDCgMju6D1HV8fWe298okbih1JEwaPrhxQTfQtSMAYJDD6wsTVhgvocPenUWUqU3grSGSs9hs9s19Y4gbkitj15cbu8BxmmMvTkW8sxAosLVvFHeokFxALtNiw4y4dA3i8XQtLFCi6KDj51eHYR7S5zFBXreFMCDwPU99ZjxDq1G9dFK8oN74QLPM7qHjNYh7SCtmzWDsZhyTU6t31BSCHPLa3L8pAcpqSK29DBvm6bc1VCU2LJTdVbPbNjn2vG&message=3CQ2BA41fzRS3uv7v&display=utf8`

### Phantom Version

/v1/

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by MakingStan** [comment]
Alot of errors are simply just "error when ....", they seriously need to fix this.

**Reply by airyland** [reply]
Seems they don't care.

**Reply by adamdelphantom** [reply]
@airyland Hello, sorry for the delay here! It just takes time for us to be able to investigate. Will get back to you with more info as soon as possible. Thank you for understanding!
